src/errors/limit-error.ts

```diff
@@ -0,0 +1,13 @@
+/**
```

Comment:
this file in general could have its name changed to `types.ts` and have the limit error type and any upcoming error type that might arise in the future